R/commonOrdinalRestrictions.R

```diff
@@ -0,0 +1,1190 @@
+  ihtTable$addColumnInfo(name = "test",     title = gettext("Test"),            type = "string", combine = TRUE)
+  ihtTable$addColumnInfo(name = "stat",     title = gettext("Test Statistic"),  type = "number")
+  ihtTable$addColumnInfo(name = "pvalue",   title = gettext("p"),               type = "pvalue")
+
+  result <- try(.aorCalculateIHT(model[["fit"]]))
+
+  if(!isTryError(result)) {
+    result[["typeName"]] <- gettextf("Type %s", result[["type"]])
+    ihtTable$setData(result)
+    ihtTable$addCitation(c(
+      "Silvapulle, M. J. & Sen, P. K. (2005). Constrained statistical inference: Order, inequality, and shape constraints. Hoboken, NJ: Wiley.",
+      "Vanbrabant, L. & Rosseel, Y. (2020). Restricted statistical estimation and inference for linear models. http://restriktor.org"
+    ))
+    # for some reason, this takes forever
+    # .aorAddInformativeHypothesisTestFootnotes(ihtTable, result)
+  } else {
+    message <- .aorExtractErrorMessageSoft(result)
+    ihtTable$setError(gettextf("Could not compute the informative hypothesis tests. Error message: %s", message))
+  }
+
+}
+
+.aorAddInformativeHypothesisTestFootnotes <- function(table, result) {
+  for(i in seq_len(nrow(result)))
+    table$setRowName(rowIndex = i, newName = sprintf("row%s", i))
+
+  # hypothesis types
+  .aorInformativeHypothesisTestFootnoteHelper(
+    table   = table,
+    rows    = which.min(result[["type"]] == "classical"),
+    colName = "typeName",
+    message = gettextf("H%1$s: All equality restrictions are active (==), H%2$s: At least one equality restriction is violated.", "\u2080", "\u2081")
+    )
+
+  .aorInformativeHypothesisTestFootnoteHelper(
+    table   = table,
+    rows    = which.min(result[["type"]] == "global"),
+    colName = "typeName",
+    message = gettextf("H%1$s: All parameters are restricted to be equal (==), H%2$s: At least one inequality restriction is strictly true (>).", "\u2080", "\u2081")
+  )
+
+  .aorInformativeHypothesisTestFootnoteHelper(
+    table   = table,
+    rows    = which.min(result[["type"]] == "A"),
+    colName = "typeName",
+    message = gettextf("H%1$s: All restrictions are equalities (==), H%2$s: At least one inequality restriction is strictly true (>).", "\u2080", "\u2081")
+  )
+
+  .aorInformativeHypothesisTestFootnoteHelper(
+    table   = table,
+    rows    = which.min(result[["type"]] == "B"),
+    colName = "typeName",
+    message = gettextf("H%1$s: All restrictions hold, H%2$s: At least one restriction is violated.", "\u2080", "\u2081")
+  )
+
+  .aorInformativeHypothesisTestFootnoteHelper(
+    table   = table,
+    rows    = which.min(result[["type"]] == "C"),
+    colName = "typeName",
+    message = gettextf("H%1$s: At least one restriction is false or active (==), H%2$s: All restrictions are strictly true (>).", "\u2080", "\u2081")
+  )
+
+}
+
+.aorInformativeHypothesisTestFootnoteHelper <- function(table, rows, colName, message) {
+  for(row in rows) {
+    table$addFootnote(
```

Comment:
Also, instead of adding a footnote for each row...
Can't you just collect all the rowNames together and pass it once?

Reply:
Thanks, I can do that.

I rewrote the function according to your feedback, but it still does not work in JASP.

Just as a note, while playing with the footnotes in R, I noticed that running the following in an interactive R session:

```r
df <- data.frame(x = c(1, 2), b = c("a", "b"))
rownames(df) <- c("r1", "r2")

table <- createJaspTable()
table$setData(df)
table$addFootnote(message = "Some footnote", colNames = "x")
```
returns `toString failed because of: std::exception`. Can this be related?

Reply:
Maybe there is some bug there...